software/controller/lib/hal/spi_chain.tpp

```diff
@@ -222,14 +207,42 @@ void DaisyChain<MaxSlaves>::TransmitNextCommand() {
   }
 
   if(data_to_send){
```

Comment:
only the following block does anything with `spi_`, which makes me think that this function is likely asking to be broken (at least) in half.
In fact, other than the setup functions, the rx handler and `SendDataWithBusyWait`, this is the only place the backend is accessed. So it seems to me the 2 "domains" are already almost completely factored away from each other.